grails-app/domain/org/pih/warehouse/inventory/CycleCountCandidateInProgress.groovy

```diff
@@ -0,0 +1,84 @@
+package org.pih.warehouse.inventory
+
+import org.pih.warehouse.core.Location
+import org.pih.warehouse.core.Tag
+import org.pih.warehouse.core.User
+import org.pih.warehouse.product.Product
+
+
+class CycleCountCandidateInProgress {
+
+    String id
+
+    CycleCountRequest cycleCountRequest
```

Comment:
actually all the fields below come from the `cycleCountRequest`. This `cycleCountRequest` is quite redundant because e.g. `facility == cycleCountRequest.facility` etc., but I've included the `cycleCountRequest` separately here, not to have to refactor frontend, as the previous API returned the `cycleCountRequest` and we were looking for ids of those.
I felt like it was a bit risky to touch the frontend in that matter to push the draft, so I left it like that for now.
It doesn't hurt the performance much, so if you are fine with it for now, let's leave it, otherwise, I can do the appropriate refactor on frontend on Monday.

Reply:
hm interesting. Is the idea that we want this object to be distinct from CycleCountRequest? So if a field is added to CycleCountRequest, we aren't forced to also add it to CycleCountCandidateInProgress. I can see the positives and negatives of this. It seems simpler to just shove the request in there so that we don't need to remember to modify the class or the sql if we want to return an additional field, but then again I do like letting this be its own thing. But maybe all it's doing is hiding the fact that it's dependent on CycleCountRequest by moving that to the sql and so it'd be better to be more visible about the dependency.

These are just some loose thoughts. I'd be curious to hear what the others think